runtime/expr/cast.go

```diff
@@ -290,3 +292,21 @@ func (c *casterNamedType) Eval(ectx Context, this *zed.Value) *zed.Value {
 	}
 	return ectx.NewValue(typ, val.Bytes())
 }
+
+type casterTypeValue struct {
+	zctx *zed.Context
+}
+
+func (c *casterTypeValue) Eval(ectx Context, val *zed.Value) *zed.Value {
+	if val.Type().ID() != zed.IDString {
```

Comment:
`type(<int64>)` should succeed.
```suggestion
	id := val.Type().ID()
	if id == zed.IDType {
		return return val
	}
	if id != zed.IDString {
```